src/test/java/de/fraunhofer/aisec/cpg/enhancements/variableResolution/VariableResolverCppTest.java

```diff
@@ -0,0 +1,433 @@
+import de.fraunhofer.aisec.cpg.graph.CatchClause;
+import de.fraunhofer.aisec.cpg.graph.CompoundStatement;
+import de.fraunhofer.aisec.cpg.graph.DeclaredReferenceExpression;
+import de.fraunhofer.aisec.cpg.graph.Expression;
+import de.fraunhofer.aisec.cpg.graph.FieldDeclaration;
+import de.fraunhofer.aisec.cpg.graph.ForStatement;
+import de.fraunhofer.aisec.cpg.graph.FunctionDeclaration;
+import de.fraunhofer.aisec.cpg.graph.IfStatement;
+import de.fraunhofer.aisec.cpg.graph.Literal;
+import de.fraunhofer.aisec.cpg.graph.MemberExpression;
+import de.fraunhofer.aisec.cpg.graph.MethodDeclaration;
+import de.fraunhofer.aisec.cpg.graph.Node;
+import de.fraunhofer.aisec.cpg.graph.ParamVariableDeclaration;
+import de.fraunhofer.aisec.cpg.graph.RecordDeclaration;
+import de.fraunhofer.aisec.cpg.graph.TranslationUnitDeclaration;
+import de.fraunhofer.aisec.cpg.graph.VariableDeclaration;
+import de.fraunhofer.aisec.cpg.helpers.NodeComparator;
+import de.fraunhofer.aisec.cpg.helpers.SubgraphWalker;
+import de.fraunhofer.aisec.cpg.helpers.Util;
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+// Todo VariableResolverPass 23 Failed, 5 Passed
+
+@Disabled(
+    "Until changing variable resolution to ScopeManager. Then in detail disable the tests that need specific fixes")
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class VariableResolverCppTest {
+
+  private RecordDeclaration externalClass;
+  private FieldDeclaration externVarName;
+  private FieldDeclaration externStaticVarName;
+
+  private RecordDeclaration outerClass;
+  private FieldDeclaration outerVarName;
+  private FieldDeclaration outerStaticVarName;
+  private FieldDeclaration outerImpThis;
+
+  private RecordDeclaration innerClass;
+  private FieldDeclaration innerVarName;
+  private FieldDeclaration innerStaticVarName;
+  private FieldDeclaration innerImpThis;
+  private FieldDeclaration innerImpOuter;
+
+  private FunctionDeclaration main;
+
+  private MethodDeclaration outer_function1;
+  private List<ForStatement> forStatements;
+  private MethodDeclaration outer_function2;
+  private MethodDeclaration outer_function3;
+  private MethodDeclaration outer_function4;
+  private MethodDeclaration outer_function5;
+
+  private MethodDeclaration inner_function1;
+  private MethodDeclaration inner_function2;
+
+  private Map<String, Expression> callParamMap = new HashMap<>();
+
+  @BeforeAll
+  public void initTests() throws ExecutionException, InterruptedException {
+    final String topLevelPath = "src/test/resources/variablesExtended/";
+    List<String> fileNames =
+        Arrays.asList("scope_variables.cpp", "external_class.cpp", "external_class.h");
```

Comment:
Why is the header file included as a translation unit? It should not be and rather `loadIncludes` should be set to true

Reply:
true

Reply:
No, you have to load all cpp files, but the header files will be loaded through `loadIncludes`. This simulates the usual behaviour if you would compile the program, you would only specify the cpp files as translation units.